python/ql/lib/semmle/python/Files.qll

```diff
@@ -144,34 +159,8 @@ class Folder extends Container, @folder {
  * A container is an abstract representation of a file system object that can
  * hold elements of interest.
  */
-abstract class Container extends @container {
-  Container getParent() { containerparent(result, this) }
-
-  /**
-   * Gets a textual representation of the path of this container.
-   *
-   * This is the absolute path of the container.
-   */
-  string toString() { result = this.getAbsolutePath() }
-
-  /**
-   * Gets the relative path of this file or folder from the root folder of the
-   * analyzed source location. The relative path of the root folder itself is
-   * the empty string.
-   *
-   * This has no result if the container is outside the source root, that is,
-   * if the root folder is not a reflexive, transitive parent of this container.
-   */
-  string getRelativePath() {
-    exists(string absPath, string pref |
-      absPath = this.getAbsolutePath() and sourceLocationPrefix(pref)
-    |
-      absPath = pref and result = ""
-      or
-      absPath = pref.regexpReplaceAll("/$", "") + "/" + result and
-      not result.matches("/%")
-    )
-  }
+class Container extends Impl::Container {
+  Container getParent() { result = this.getParentContainer() }
```

Comment:
Looks like we have an extra class predicate here that we should clean up (users should just use `getParentContainer` instead)?

Reply:
Yeah, I didn't want to change things so I just left it in.

Reply:
That is fine, we can clean it up later :-)